pkg/ring/kv/multi.go

```diff
@@ -0,0 +1,357 @@
+package kv
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/uber-go/atomic"
+
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+var (
+	primaryStoreGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "consul_multikv_primary_store",
+		Help: "Selected primary KV store",
+	}, []string{"store"})
+
+	mirrorEnabled = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "consul_multikv_mirror_enabled",
+		Help: "Is mirroring to secondary store enabled",
+	})
+
+	mirrorWrites = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "consul_multikv_mirror_writes_total",
+		Help: "Number of mirror-writes to secondary store",
+	})
+
+	mirrorFailures = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "consul_multikv_mirror_write_errors_total",
+		Help: "Number of failures to mirror-write to secondary store",
+	})
+)
+
+func init() {
+	prometheus.MustRegister(primaryStoreGauge, mirrorEnabled, mirrorWrites, mirrorFailures)
+}
+
+// MultiConfig is a configuration for MultiClient.
+type MultiConfig struct {
+	Primary   string `yaml:"primary"`
+	Secondary string `yaml:"secondary"`
+
+	MirrorEnabled bool          `yaml:"mirror-enabled"`
```

Comment:
To keep consistency, YAML config name should have `_` instead of `-` (same comment applies to `MirrorTimeout`)